package/gluon-node-info/src/respondd.c

```diff
@@ -0,0 +1,144 @@
+  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+
+
+#include <respondd.h>
+
+#include <json-c/json.h>
+#include <libgluonutil.h>
+
+#include <uci.h>
+
+#include <stdlib.h>
+#include <string.h>
+
+
+static struct uci_section * get_first_section(struct uci_package *p, const char *type) {
+        struct uci_element *e;
+        uci_foreach_element(&p->sections, e) {
+                struct uci_section *s = uci_to_section(e);
+                if (!strcmp(s->type, type))
+                        return s;
+        }
+
+        return NULL;
+}
+
+static const char * get_first_option(struct uci_context *ctx, struct uci_package *p, const char *type, const char *option) {
+        struct uci_section *s = get_first_section(p, type);
+        if (s)
+                return uci_lookup_option_string(ctx, s, option);
+        else
+                return NULL;
+}
+
+static struct json_object * get_number(struct uci_context *ctx, struct uci_section *s, const char *name) {
+        const char *val = uci_lookup_option_string(ctx, s, name);
+        if (!val || !*val)
+                return NULL;
+
+        char *end;
+        double d = strtod(val, &end);
+        if (*end)
+                return NULL;
+
+        return json_object_new_double(d);
+}
+
+static struct json_object * get_location(struct uci_context *ctx, struct uci_package *p) {
+        struct uci_section *s = get_first_section(p, "location");
+        if (!s)
+                return NULL;
+
+        const char *share = uci_lookup_option_string(ctx, s, "share_location");
+        if (!share || strcmp(share, "1"))
+                return NULL;
+
+        struct json_object *ret = json_object_new_object();
+
+        struct json_object *latitude = get_number(ctx, s, "latitude");
+        if (latitude)
+                json_object_object_add(ret, "latitude", latitude);
+
+        struct json_object *longitude = get_number(ctx, s, "longitude");
+        if (longitude)
+                json_object_object_add(ret, "longitude", longitude);
+
+        struct json_object *altitude = get_number(ctx, s, "altitude");
+        if (altitude)
+                json_object_object_add(ret, "altitude", altitude);
+
+        return ret;
+}
+
+static struct json_object * get_owner(struct uci_context *ctx, struct uci_package *p) {
+        const char *contact = get_first_option(ctx, p, "owner", "contact");
+        if (!contact || !*contact)
+                return NULL;
+
+        struct json_object *ret = json_object_new_object();
+        json_object_object_add(ret, "contact", gluonutil_wrap_string(contact));
+        return ret;
+}
+
+static struct json_object * get_system(struct uci_context *ctx, struct uci_package *p) {
+        struct json_object *ret = json_object_new_object();
+
+        const char *role = get_first_option(ctx, p, "system", "role");
+        if (role && *role)
+                json_object_object_add(ret, "role", gluonutil_wrap_string(role));
```

Comment:
Why does `get_system()` behave differently from `get_owner()` and `get_location()` in that it returns an empty JSON object instead of NULL when the corresponding setting isn't set?